### PR TITLE
Constant propagation changes

### PR DESCRIPTION
**Pull request by eellison**
- remove loop node that is guaranteed not to execute
- remove extra loop outputs that are no longer needed

- if we are inlining an if node, only run constant propagation on the block that will execute

- remove the recurse argument since we only expose the Graph Constant Propagation and it's not used

This also includes  a few extra hooks to python_ir that I think make it a little be easier to test graph conditions from python.

### CODE REVIEW

File: torch/csrc/jit/passes/constant_propagation.cpp [hunk 1]
```diff
@@ -126,53 +143,78 @@ bool removeExtraNodeOutputs(Node* n) {
   return initial_outputs != true_block->outputs().size();
 }
 
-void ConstantPropagation(Block* block, const AliasDb& aliasDb, bool recurse);
+// remove extra outputs from the node
+void removeExtraLoopOutputs(Node* node) {
+  auto loop_body = node->blocks().at(0);
+  auto loop_input_offset = 2; // offset of loop carried deps in input list
```
Comment (driazati) [inline torch/csrc/jit/passes/constant_propagation.cpp]: pool this and the other one into a `static const`

File: torch/csrc/jit/python_ir.cpp [hunk 2]
```diff
@@ -68,6 +68,35 @@ std::ostream& printPyObject(std::ostream& out, const THPObjectPtr& obj) {
   }
 }
 
+std::vector<Node*> findAllNodes(Block* block, Symbol kind) {
```
Comment (driazati) [inline torch/csrc/jit/python_ir.cpp]: I don't think these are really necessary to add if it's just for the test, this functionality can be done per-case via the Python bindings for `Graph` and `Node`

Comment (eellison) [reply torch/csrc/jit/python_ir.cpp]: What do you mean per-case via the bindings for Graph and Node? 

These will also be used in other tests, and have been a common pattern I've needed & couldn't easily express so I used expect files.

Comment (eellison) [reply torch/csrc/jit/python_ir.cpp]: If I don't add it as a method to Graph / Node then you can't chain method calls, so something like the following becomes difficult to write:
`
self.assertTrue(graph.findNode("prim::Loop").outputsSize() == 2)
`

File: torch/csrc/jit/passes/constant_propagation.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/passes/constant_propagation.cpp]: I can change it - fwiw I was following the style of similar code that exists in dead_code_elimination & uses auto

**Review by driazati — APPROVED** [review]

**Pull request closed without merging**